pyuvdata/uvfits.py

```diff
@@ -449,8 +454,18 @@ def read_uvfits(self, filename, antenna_nums=None, antenna_names=None,
             # stuff in columns
             ant_names = ant_hdu.data.field('ANNAME').tolist()
             self.antenna_names = []
-            for name in ant_names:
-                self.antenna_names.append(name.replace('\x00!', ''))
+            for ant_ind, name in enumerate(ant_names):
+                # Sometimes CASA writes antnames as bytes not strings.
+                # If the ant name is shorter than 8 characters, the trailing
+                # characters may not be able to be decoded.
+                # So we just ignore any non-ascii bytes in the decode.
+                if isinstance(name, bytes):
+                    ant_name_str = None
+                    ant_name_str = str(name.decode('utf-8', 'ignore'))
+                else:
+                    ant_name_str = name
+                self.antenna_names.append(ant_name_str.replace(
+                    '\x00', '').replace('\x07', '').replace('!', ''))
```

Comment:
There's probably a better way to do this with a regex. We just want to remove anything that's not alphanumeric.

Reply:
`antenna_names.append(re.sub(r'[^a-zA-Z0-9 -]', '', name)) `
This gets rid of *all* alphanumeric
@plaplant is a good regex person to double check.

Reply:
Adam means it gets rid of all _non_-alphanumeric

Reply:
Yes, as written, @adampbeardsley's regex allows only for alphanumeric characters, spaces, and dashes. Are there other (ASCII) characters that we should permit? I know that HERA doesn't need anything else, but I didn't know if other telescopes used things like underscores or periods or whatnot.

Whatever we decide on, we should probably document somewhere. I quickly went through the AIPS memo on uvfits, and I couldn't find anything specifying which characters were permitted in antenna names. If we're going to be more restrictive than "valid ASCII", we might want to say so (and perhaps globally enforce that somewhere in the `antenna_names` attribute).

Reply:
Ok, we can go back to what I had before, which only strips non-printing ascii characters and exclamation points, all of these I've established to not be desired via roundtrips with CASA measurement sets and uvfits

Reply:
@plaplant I feel comfortable removing the non-printing characters, which I think are only present because the fields are fixed width. The exclamation points appear in VLA uvfits files and when CASA converts them to measurement sets it strips the exclamation points. I find it hard to believe the VLA wants exclamation points in their antenna names, so I think striping them makes sense. If we limit ourselves to just striping those do you think we need to add documentation somewhere (and if so, where?).

Reply:
Thanks very much for digging into the round-tripping. If that's the case, I think the comment you have in place above the string handling is sufficient. As you say, the exclamation points following non-printing characters is probably a CASA artifact.